File: apps/android/PrismEra/app/src/main/java/com/prismera/app/network/AuthService.kt
package com.prismera.app.network

import com.prismera.app.data.model.AuthResponse
import com.prismera.app.data.model.AuthToken
import com.prismera.app.data.model.User
import com.prismera.app.data.local.PreferenceManager

class AuthService private constructor() {
    companion object {
        val instance: AuthService by lazy { AuthService() }
        fun getInstance(): AuthService = instance
    }
    
    suspend fun login(email: String, password: String): Result<AuthResponse, ApiException> {
        val response = ApiClient.getInstance().post<AuthResponse, LoginRequest>(
            "/auth/login",
            LoginRequest(email, password)
        )
        // Store token securely
        PreferenceManager.getInstance().saveAuthToken(response.token)
        // Cache user for offline access
        CacheManager.getInstance().set(response.user, "current_user")
        
        return Result.success(response)
    }
    
    suspend fun register(name: String, email: String, password: String): Result<AuthResponse, ApiException> {
        val response = ApiClient.getInstance().post<AuthResponse, RegisterRequest>(
            "/auth/register",
            RegisterRequest(name, email, password)
        )
        PreferenceManager.getInstance().saveAuthToken(response.token)
        CacheManager.getInstance().set(response.user, "current_user")
        return Result.success(response)
    }
    
    suspend fun logout(): Result<Unit, ApiException> {
        try {
            ApiClient.getInstance().post<Unit, EmptyBody>("/auth/logout", EmptyBody())
        } finally {
            PreferenceManager.getInstance().clearAuthToken()
            CacheManager.getInstance().remove("current_user")
        }
        return Result.success(Unit)
    }
    
    suspend fun validateSession(): Result<User?, ApiException> {
        return try {
            val user = ApiClient.getInstance().get<User>("/auth/me")
            Result.success(user)
        } catch (e: ApiException.Unauthorized) {
            Result.success(null)
        }
    }
    
    suspend fun refreshToken(): Result<AuthToken?, ApiException> {
        return try {
            val token = ApiClient.getInstance().post<AuthToken, EmptyBody>("/auth/refresh", EmptyBody())
            PreferenceManager.getInstance().saveAuthToken(token)
            Result.success(token)
        } catch (e: Exception) {
            Result.success(null)
        }
    }
}

// Request models
data class LoginRequest(val email: String, val password: String)
data class RegisterRequest(val name: String, val email: String, val password: String)

File: apps/ios/PrismEra/Services/AuthService.swift
import Foundation

class AuthService {
    static let shared = AuthService()
    private let apiClient = APIClient.shared

    private init() {}

    // MARK: - Authentication

    func login(email: String, password: String) async throws -> AuthResponse {
        let body = LoginRequest(email: email, password: password)
        return try await apiClient.post("/auth/login", body: body)
    }

    func register(name: String, email: String, password: String) async throws -> AuthResponse {
        let body = RegisterRequest(name: name, email: email, password: password)
        return try await apiClient.post("/auth/register", body: body)
    }

    func logout() async throws {
        try await apiClient.post("/auth/logout", body: EmptyBody())
    }

    func validateSession() async throws -> User? {
        do {
            let response: User = try await apiClient.get("/auth/me")
            return response
        } catch APIError.httpError(let statusCode) where statusCode == 401 {
            return nil
        }
    }

    func refreshToken() async throws -> AuthToken? {
        do {
            let response: AuthToken = try await apiClient.post("/auth/refresh", body: EmptyBody())
            return response
        } catch {
            return nil
        }
    }

    // MARK: - Password Reset

    func requestPasswordReset(email: String) async throws {
        let body = EmailRequest(email: email)
        try await apiClient.post("/auth/forgot-password", body: body)
    }

    func resetPassword(token: String, newPassword: String) async throws {
        let body = ResetPasswordRequest(token: token, newPassword: newPassword)
        try await apiClient.post("/auth/reset-password", body: body)
    }

    // MARK: - OAuth (Clerk Integration)

    func handleOAuthCallback(code: String, state: String) async throws -> AuthResponse {
        let body = OAuthCallbackRequest(code: code, state: state)
        return try await apiClient.post("/auth/oauth/callback", body: body)
    }
}

// MARK: - Request Models

struct LoginRequest: Encodable {
    let email: String
    let password: String
}

struct RegisterRequest: Encodable {
    let name: String
    let email: String
    let password: String
}

struct EmailRequest: Encodable {
    let email: String
}

struct ResetPasswordRequest: Encodable {
    let token: String
    let newPassword: String
}

struct OAuthCallbackRequest: Encodable {
    let code: String
    let state: String
}
